libraries/core/src/descriptor/mod.rs

```diff
@@ -176,6 +178,25 @@ pub fn source_is_url(source: &str) -> bool {
     source.contains("://")
 }
 
+pub fn source_is_path(source: &str, working_dir: &Path) -> Result<PathBuf> {
```

Comment:
The "is" in the method name sounds like this returns a `bool`. Maybe rename this to `canoncialize_path` or `resolve_path`?

Reply:
Agreed. I'll go with `resolve_path`. Thanks!